docs/src/main/asciidoc/security-authorization-of-web-endpoints-reference.adoc

```diff
@@ -8,55 +8,68 @@ https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 include::_attributes.adoc[]
 :categories: security,web
 
-Quarkus has an integrated pluggable web security layer. If security is enabled all HTTP requests will have a permission
-check performed to make sure they are allowed to continue.
+Quarkus has an integrated pluggable web security layer. If security is enabled, all HTTP requests will have a permission check performed to make sure they are allowed to continue.
+This means you can not use `@PermitAll` to open up a path if the path has been blocked by using the `quarkus.http.auth.` configuration.
 
-NOTE: Configuration authorization checks are executed before any annotation-based authorization check is done, so both
-checks have to pass for a request to be allowed. This means you cannot use `@PermitAll` to open up a path if the path has
-been blocked using `quarkus.http.auth.` configuration. If you are using JAX-RS you may want to consider using the
-`quarkus.security.jaxrs.deny-unannotated-endpoints` or `quarkus.security.jaxrs.default-roles-allowed` to set default security
-requirements instead of HTTP path level matching, as these properties can be overridden by annotations on an individual
-endpoint.
+[NOTE]
+====
+If you are using JAX-RS, consider using `quarkus.security.jaxrs.deny-unannotated-endpoints` or `quarkus.security.jaxrs.default-roles-allowed` to set default security requirements instead of HTTP path level matching because annotations can override these properties on an individual endpoint.
+====
 
-Authorization is based on user roles that are provided by the security provider.
+Authorization is based on user roles that the security provider provides.
 To customize these roles, a `SecurityIdentityAugmentor` can be created, see
 xref:security-customization.adoc#security-identity-customization[Security Identity Customization].
 
-== Authorization using Configuration
+== Authorization using configuration
+
+Permissions are defined in the Quarkus configuration using permission sets, with each permission set specifying a policy for access control.
 
-The default implementation allows you to define permissions using config in `application.properties`. An example
-config is shown below:
+.{project-name} policies summary
+|===
+s| Built-in policy s| Description
+s| `deny` | This policy denies all users.
+s| `permit` | This policy permits all users.
+s| `authenticated` | This policy permits only authenticated users.
+|===
 
+It is possible to define role-based policies that will only allow users with the specified roles to access the resources.
+
+.Example of a role-based policy
 [source,properties]
 ----
-quarkus.http.auth.policy.role-policy1.roles-allowed=user,admin                      <1>
+quarkus.http.auth.policy.role-policy1.roles-allowed=user,admin                  <1>
+----
+<1> This defines a role-based policy that allows users with the `user` and `admin` roles.
+This is referenced by permission sets in the next snippet.
```

Comment:
Generally, to avoid the linguistic equivalent of a null-pointer error, specify the object following "this". 
```suggestion
This <object> is referenced by permission sets in the next snippet.
```

Reply:
Will apply this:
Such a custom policy can be referenced by permission sets just like the built-in ones, as shown in the example below.